### ISSUE TITLE
gettext builder fails with "ValueError: substring not found"

### ISSUE THREAD

**Issue by cocoatomo** [sphinx-doc/sphinx#3669]
Subject: On Python document translated into Japanese, gettext builder fails with "ValueError: substring not found."

### Problem
- gettext builder fails and do not create .pot files.

#### Procedure to reproduce the problem
```
$ git clone https://github.com/python/cpython.git
$ cd cpython
$ git remote add intl https://github.com/python-doc-ja/cpython-doc-intl.git

$ git clone https://github.com/python-doc-ja/py36-locale.git Doc/locale

$ git fetch intl
$ git checkout -t intl/intl-3.6

$ cd Doc
$ make gettext
```

ref. https://github.com/python-doc-ja/python-doc-ja/wiki/HowToUpdateOriginal

#### Error logs / results
When building with sphinx 1.5.5,
```
sphinx-build -E -b gettext -D gettext_compact=0  -w warnings.txt -d build/.doctrees . locale/pot
Running Sphinx v1.5.5
building [gettext]: targets for 6 template files
reading templates... [100%] /Users/tomohiko/Dropbox/MyWorks/Python/cpython/Doc/tools/templates/indexcontent.html                                                                                 
building [gettext]: targets for 465 source files that are out of date
updating environment: 465 added, 0 changed, 0 removed
reading sources... [100%] whatsnew/index                                                                                                                                                         
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] whatsnew/index                                                                                                                                                          
writing message catalogs... [  0%] c-api/import                                                                                                                                                  
Exception occurred:
  File "/usr/local/lib/python2.7/site-packages/sphinx/builders/gettext.py", line 174, in should_write
    old_header_index = old_content.index('"POT-Creation-Date:')
ValueError: substring not found
The full traceback has been saved in /var/folders/k5/cbwycdyj57j67yx2gdbyzz0w0000gn/T/sphinx-err-SGs3SP.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [gettext] Error 1
```
[sphinx-err-SGs3SP.txt](https://github.com/sphinx-doc/sphinx/files/956990/sphinx-err-SGs3SP.txt)
(Note: Its extension is renamed to txt.)

When building with sphinx 1.6b1,
```
sphinx-build -E -b gettext -D gettext_compact=0  -w warnings.txt -d build/.doctrees . locale/pot
Running Sphinx v1.6b1
building [gettext]: targets for 6 template files
reading templates... [100%] /Users/tomohiko/Dropbox/MyWorks/Python/cpython/Doc/tools/templates/indexcontent.html                                                                                 
building [gettext]: targets for 465 source files that are out of date
updating environment: 465 added, 0 changed, 0 removed
reading sources... [100%] whatsnew/index                                                                                                                                                         
/Users/tomohiko/Dropbox/MyWorks/Python/cpython/Doc/library/pyexpat.rst:872: WARNING: Footnote [#] is not referenced.
/Users/tomohiko/Dropbox/MyWorks/Python/cpython/Doc/library/stdtypes.rst:4630: WARNING: Footnote [3] is not referenced.
/Users/tomohiko/Dropbox/MyWorks/Python/cpython/Doc/library/xml.dom.minidom.rst:251: WARNING: Footnote [#] is not referenced.
/Users/tomohiko/Dropbox/MyWorks/Python/cpython/Doc/library/xml.etree.elementtree.rst:1195: WARNING: Footnote [#] is not referenced.
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] whatsnew/index                                                                                                                                                          
writing message catalogs... [  0%] c-api/import                                                                                                                                                  
Exception occurred:
  File "/usr/local/lib/python2.7/site-packages/sphinx/builders/gettext.py", line 199, in should_write
    old_header_index = old_content.index('"POT-Creation-Date:')
ValueError: substring not found
The full traceback has been saved in /var/folders/k5/cbwycdyj57j67yx2gdbyzz0w0000gn/T/sphinx-err-opSFuF.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [gettext] Error 1
```
[sphinx-err-opSFuF.txt](https://github.com/sphinx-doc/sphinx/files/956988/sphinx-err-opSFuF.txt)
(Note: Its extension is renamed to txt.)

#### Expected results
gettext builder finish successfully.

### Reproducible project / your project
- https://github.com/python-doc-ja/cpython-doc-intl
- https://github.com/python-doc-ja/py36-locale

### Environment info
- OS: macOS Sierra 10.12.4
- Python version: 3.6.0
- Sphinx version: 1.5.5, 1.6b1

**Comment by cocoatomo** [comment]
If skipping the command `git clone https://github.com/python-doc-ja/py36-locale.git Doc/locale`, gettext builder finish successfully.
Therefore, py36-locale may relate to this failure.

**Comment by tk0miya** [comment]
It seems your pot files are written after generated.
Now gettext builder expects `POT-Creation-Date:` header is defined. I'll fix it later.

**Comment by tk0miya** [comment]
Fixed at #3671.
Thank you for reporting!
